### PR TITLE
1204 preview versions

### PR DESCRIPTION
**Pull request by MattElcock**
### What is the context of this PR?
Fixes this bug sarah found:

> when PM approving this, I published a GB form and a NI one. When using go launcher the surveys displayed correctly, however the logos both said NI. We need to check the right theme is being carried through when publishing

### How to review 
Ensure tests pass and the bug is no longer there. Test alongside this branch of the register: 1204-preview-versions

### CODE REVIEW

File: eq-author/src/App/publish/PublishPage.js [hunk 1]
```diff
@@ -92,22 +92,36 @@ LabeledCheckbox.propTypes = {
...
+  const handleThemeSelect = (themeCheckbox, label) => {
+    const isSelected = themeCheckbox.value;
+    let variantsArray = variants;
+
+    if (isSelected) {
+      setVariants([...variantsArray, { theme: label, formType: null }]);
+    } else {
+      variantsArray = variantsArray.filter(variant => variant.theme !== label);
+      setVariants([...variantsArray]);
+    }
```
Comment (heofs) [inline eq-author/src/App/publish/PublishPage.js]: You could reduce this to 
```js
    if (isSelected) {
      setVariants([...variants, { theme: label, formType: null }]);
    } else {
      setVariants(variants.filter(variant => variant.theme !== label));
    }
```

File: eq-author/src/App/publish/PublishPage.js [hunk 2]
```diff
@@ -92,22 +92,36 @@ LabeledCheckbox.propTypes = {
...
   };
 
   const handleInputChange = event => {
-    setFormTypes({
-      ...formTypes,
-      [event.name]: event.value,
-    });
+    const theme = event.name;
+    const formType = event.value;
+    let variantsArray = [];
```
Comment (heofs) [inline eq-author/src/App/publish/PublishPage.js]:
```suggestion
    const variantsArray = [];
```

File: eq-author-api/schema/resolvers/base.js [hunk 3]
```diff
@@ -604,22 +603,48 @@ const Resolvers = {
...
+          surveyId,
+          formType: variant.formType,
+          variants: [
+            {
+              language: "en",
+              theme: themeLookup[variant.theme],
+              // eslint-disable-next-line camelcase
+              author_id: questionnaireId,
+            },
+          ],
```
[the remark this answers is not in the document]
Comment (MattElcock) [reply eq-author-api/schema/resolvers/base.js]: No, when we move to implementing multiple languages, variants will have more objects inside.

**Comment by MattElcock** [comment]
> Code looks good and tests looks good! Just had couple of suggestions.
> 
> Also, is `variants` what eQ is calling it? Or would they use another word to describe a questionnaire with a different theme?

Variants isn't a thing in eQ, it's so that the registry can propperly build the different language and theme variants for eQ to collect.

File: eq-author-api/schema/resolvers/base.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (heofs) [reply eq-author-api/schema/resolvers/base.js]: Okay, thanks for clarification :)

**Pull request merged**